music21/musicxml/xmlToM21.py

```diff
@@ -2575,16 +2584,7 @@ def parse(self):
 
         if self.useVoices is True:
             for v in self.stream.iter().voices:
-                if v:  # do not bother with empty voices
-                    # the musicDataMethods use insertCore, thus the voices need to run
-                    # coreElementsChanged
-                    v.coreElementsChanged()
-                    # Fill mid-measure gaps, and find end of measure gaps by ref to measure stream
-                    # https://github.com/cuthbertlab/music21/issues/444
-                    v.makeRests(refStreamOrTimeRange=self.stream,
```

Comment:
This is the crux of this PR -- I agree this is a useful change, all discussion on #1636 was in agreement that we want to make this change and only account for Finale's abuse of the spec, it just stalled out on code review comments. Nice job keeping this PR small.